test/e2e/main_test.go

```diff
@@ -211,17 +212,19 @@ func testInstall(
 	// create a directory used to store the @testName container and resource logs
 	testOutputDir := filepath.Join(df.LoggingPath, testName)
 	err := os.Mkdir(testOutputDir, 0777)
-	assert.NoError(t, err, "creating the test case output directory should produce no error")
+	require.NoError(t, err, "creating the test case output directory should produce no error")
 
 	// randomize the namespace to avoid existing namespaces
 	rand.Seed(time.Now().UnixNano())
 	namespace := df.NamespacePrefix + "-" + strconv.Itoa(rand.Intn(50))
 
 	deployerCtx, err := df.NewDeployerCtx(repo, tag, namespace, testOutputDir, targetPods, spec)
-	assert.NoError(t, err, "creating a new deployer context should produce no error")
+	require.NoError(t, err, "creating a new deployer context should produce no error")
+	require.NotNil(t, deployerCtx, "the DeployerCtx object should not be nil")
```

Comment:
Do we really need both? I feel like deployerCtx should never be non-nil if there's not an error, and it should never be set if there is, so the `require.NoError` should be sufficient.

Reply:
Hmm not really, I'll remove it.